test/model/game_state_test.dart

```diff
@@ -134,6 +134,26 @@ void main() {
     });
   });
 
+  group('View Discard Pile', () {
+    test('should show the same card after viewing discard pile is finished', () {
+      const plunderAWreck = PlunderAWreck();
+      const aRivalShip = ARivalShip();
+      const portFees = PortFees();
+      final GameState state = makeGameState(deck: [plunderAWreck, aRivalShip, portFees]);
+
+      state.nextCard();
+      state.nextCard();
+      expect(state.discard, [plunderAWreck]);
+      expect(state.currentCard, aRivalShip);
+
+      state.toggleDiscardView();
+      state.toggleDiscardView(closeWindow: true);
+
+      expect(state.discard, [plunderAWreck]);
+      expect(state.currentCard, aRivalShip);
+    });
```

Comment:
Feels a bit of an odd test, because the method we're testing doesn't go near the aspects of the state that we're verifying. Maybe something like this is all we need?:

```
    test('should show the same card after viewing discard pile is finished', () {
      final GameState state = makeGameState();
      expect(state.viewingDiscard, false);

      state.toggleDiscardView();
      expect(state.viewingDiscard, true);

      state.toggleDiscardView();
      expect(state.viewingDiscard, false);
    });
```